queue/queue.go

```diff
@@ -6,15 +6,28 @@ import (
 	"time"
```

Comment:
@ogbanugot This file should define something like this

```
func NewQueue(opts QueueOptions) Queuer {
  if opts.Type == "in-memory" {
	return memqueue.NewQueue(opts)
  }

  if opts.Type == "redis" {
	return redisqueue.NewQueue(opts)
  }

  return errors.New("invalid type")
}
```

This is what will be called in `main.go` as `queue.NewQueue(opts)`

Reply:
Makes sense, this takes care of the check for which `NewQueue()` to call, what about the check for which `NewClient()` to call, should we set a condition on the `cfg.Queue.Type` in `cmd/main.go`. In other words where does the toggle between queue types happen in `cmd/main.go`?